locations/storefinders/rio_seo.py

```diff
@@ -17,64 +15,69 @@ class RioSeoSpider(Spider):
     RioSEO is a number of related storefinders.
     https://www.rioseo.com/platform/local-pages/
 
-    To use, specify `end_point`
+    To use, specify `domain` as the API domain without any preceding "maps.".
     """
 
     dataset_attributes = {"source": "api", "api": "rio_seo"}
 
-    end_point: str = None
-    radius: int = 10000
-    limit: int = 3000
+    domain: Optional[str] = None
+    limit: int = 10000
+    radius: int = 20038
 
     def start_requests(self) -> Iterable[Request]:
-        if self.start_urls:
-            for url in self.start_urls:
-                yield JsonRequest(url=url)
-        else:
-            yield JsonRequest(url=urljoin(self.end_point, "getAutocompleteData"), callback=self.parse_autocomplete)
+        yield JsonRequest(f"https://maps.{self.domain}/api/getAutocompleteData", callback=self.parse_autocomplete)
 
     def parse_autocomplete(self, response: Response, **kwargs: Any) -> Any:
         yield JsonRequest(
-            url=urljoin(
-                self.end_point,
-                "getAsyncLocations?template=search&level=search&search={}&radius={}&limit={}".format(
-                    response.json()["data"][0], self.radius, self.limit
-                ),
-            )
+            f"https://maps.{self.domain}/api/getAsyncLocations?template=domain&level=domain&search={response.json()['data'][0]}&radius={self.radius}&limit={self.limit}"
         )
 
-    def parse(self, response, **kwargs):
+    def parse(self, response: Response, **kwargs) -> Iterable[Feature]:
         map_list = response.json()["maplist"]
-        try:
-            data = json.loads("[{}]".format(Selector(text=map_list).xpath("//div/text()").get()[:-1]))
-        except json.decoder.JSONDecodeError:
-            self.logger.warning("Could not parse response - check API output")
-            data = []
-        except TypeError:
-            data = []
+        if map_list == "":
+            return
+        map_list = map_list.removeprefix('<div class="tlsmap_list">')
+        map_list = map_list.removesuffix("</div>")
+        data = json.loads(f"[{map_list[:-1]}]")
+
+        if len(data) == self.limit:
+            self.logger.warning(f"Received {len(data)} entries, the limit may need to be raised")
 
         for location in data:
             feature = DictParser.parse(location)
-            feature["name"] = location["location_name"]
-            feature["ref"] = "{}_{}".format(location["fid"], location["lid"])
             feature["street_address"] = merge_address_lines([location["address_1"], location["address_2"]])
+            feature["ref"] = location["fid"]
```

Comment:
I'm a little uncomfortable changing the ref, I don't know why it was made of both of them, but I assume it was for a reason.

Reply:
Yeah, I'd like to know too… oh apparently I wrote that part originally 😅. I don't remember why I chose both. Maybe I was being overly cautious of uniqueness, maybe I couldn't decide which to use, or maybe See's uses that combination on the website. This time I went with only `fid` because it's the one that appears in URLs, which I think makes it a natural `ref`. I verified it is unique within each spider.